### DISCUSSION TITLE
High CPU usage

### DISCUSSION THREAD

**Discussion by burgerga** [nicolargo/glances#3043] [General]
Resolved via https://github.com/nicolargo/glances/issues/3033
Sorry for the late feedback, but its running great on proxmox now, many thanks @nicolargo!
---

**Check the bug**
Previous issues were closed as solved

**Describe the bug**
Instead of <5% CPU Usage, I have around 15-25% CPU usage for glances on a Intel Core i3-8100T (running Proxmox)

**To Reproduce**
1. Install glances on Proxmox using https://raw.githubusercontent.com/nicolargo/glancesautoinstall/master/install.sh (via https://community-scripts.github.io/ProxmoxVE/scripts?id=glances)
2. Start via systemd service (`/usr/local/bin/glances -w`)
3. Open glances website
4. Observe high CPU

**Expected behavior**
Observe low CPU

**Screenshots**
![image](https://github.com/user-attachments/assets/ec8b20f0-3fc3-457f-b511-d77ccbb2a127)

**Environement (please complete the following information)**
 - Operating System (lsb_release -a or OS name/version): 
    Proxmox: 
    ```
    Distributor ID: Debian
    Description:    Debian GNU/Linux 12 (bookworm)
    Release:        12
    Codename:       bookworm
    ```
 - Glances & psutil versions: 
    ```
    Glances version:        4.2.1
    Glances API version:    4
    PsUtil version:         5.9.4
    Log file:               /root/.local/share/glances/glances.log
    ``` 
 - How do you install Glances (Pypi package, script, package manager, source): https://raw.githubusercontent.com/nicolargo/glancesautoinstall/master/install.sh
 - Glances test: 
[glances_issue.txt](https://github.com/user-attachments/files/17848176/glances_issue.txt)

**Additional context**
I have around 30 LXCs running in proxmox, so the processlist is quite long.
How can I further debug this?

You can also [pastebin](https://pastebin.com/):

* the Glances configuration file (https://glances.readthedocs.io/en/latest/config.html#location): No config file
* the Glances log file (https://glances.readthedocs.io/en/latest/config.html#logging): 
[glances.log](https://github.com/user-attachments/files/17848309/glances.log)

**Comment by burgerga** [comment]
Well, I guess this is indeed part of the problem:
```
processcount  [OK]    0.24868s
{'pid_max': 0, 'running': 1, 'sleeping': 922, 'thread': 2347, 'total': 1091}
```
But I also get high CPU if I request only CPU and MEM use using the [gethomepage.dev glances info widget](https://gethomepage.dev/widgets/info/glances/)

**Comment by nicolargo** [comment]
Strange, Glances is now optimzed for CPU consumption (on my PC, it consume ~5% CPU).

Can you try this:

    glances --disable-plugin all --enable-plugin quicklook,cpu,mem,swap,load,processlist

**Comment by burgerga** [comment]
Then it's between 8-15%, but mostly around 10%. If I remove `processlist`, and use `top` in a different shell I can see it does 3%.

**Comment by burgerga** [comment]
Apart from opening the website, I also tried using the API by using the homepage widget, which doesn't use the `processlist` endpoint (see https://github.com/gethomepage/homepage/blob/a06964dd175fa798fa9189ba1ecd5766bbde543f/src/pages/api/widgets/glances.js) but still causes high cpu... is this triggered by accessing the API?

**Comment by nicolargo** [comment]
You can also try to limit the number of processes to display in the WebUI.

In the glances.conf file:

```
[outputs]
...
# Limit the number of processes to display (in the WebUI)
max_processes_display=25
...
```

Yes for the moment, Glances update all plugins when a call is done to the API (but limited at one update every <refresh> time). An optimisation can be discussed on this point but it is not straighforward because plugins can be linked (for example processlist and alert). I just open an issue to follow this specific point: https://github.com/nicolargo/glances/issues/3033

**Comment by burgerga** [comment]
Hmm, that doesn't seem to matter very much, it's still around 20%.

However your linked issue is what I suspected. As a temporary workaround I now started a separate glances with:

```
glances --disable-plugin all --enable-plugin cpu,mem,fs,load,uptime,sensors -p 61209 -w --disable-webui
```
and connected it to my dashboard. It's now working perfectly with 0-3% CPU, and I can still access the cpu-hungry web interface at 61208 :)

**Comment by gerroon** [comment]
How do I use the workaround with the Docker run setup?

**Reply by burgerga** [reply]
The workaround should not be needed anymore as this issue has been resolved (see https://github.com/nicolargo/glances/issues/3033) :)
